arcgis-ios-sdk-samples/Layers/Apply mosaic rule to rasters/ApplyMosaicRuleToRastersViewController.swift

```diff
@@ -0,0 +1,145 @@
+                "Center": centerRule,
+                "ByAttribute": byAttributeRule,
+                "LockRaster": lockRasterRule]
+    }()
+    
+    // MARK: Methods
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        // Check if a mosaic rule exists. If not, create one.
+        if imageServiceRaster.mosaicRule == nil {
+            imageServiceRaster.mosaicRule = AGSMosaicRule()
+        }
+        // Create a raster layer.
+        let rasterLayer = AGSRasterLayer(raster: imageServiceRaster)
+        let map = AGSMap(basemap: .topographicVector())
+        // Add raster layer as an operational layer to the map.
+        map.operationalLayers.add(rasterLayer)
+        SVProgressHUD.show(withStatus: "Loading")
+        rasterLayer.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                SVProgressHUD.dismiss()
+                self.presentAlert(error: error)
+            } else {
+                // When loaded, set the map view's viewpoint to the image service raster's center.
+                if let center = self.imageServiceRaster.serviceInfo?.fullExtent?.center {
+                    self.mapView.setViewpoint(AGSViewpoint(center: center, scale: 25000.0))
+                }
+                self.setStatus(message: "Image service raster loaded.")
+            }
+        }
+        return map
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func chooseMosaicRule(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose a mosaic rule to apply to the image service.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        mosaicRulePairs.forEach { name, rule in
+            let action = UIAlertAction(title: name, style: .default) { _ in
+                SVProgressHUD.show(withStatus: "Loading mosaic rule")
+                self.setStatus(message: "\(name) selected.")
+                self.imageServiceRaster.mosaicRule = rule
+            }
+            alertController.addAction(action)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["ApplyMosaicRuleToRastersViewController"]
+    }
+    
+    deinit {
+        // Nilify the KVO after the main view disappears.
+        drawStatusObservation = nil
+    }
```

Comment:
It seems that this is [not needed](https://stackoverflow.com/questions/48722189/do-observers-need-to-be-removed-in-swift-4-objective-c). Will need reconfirm with Phil.

Reply:
Ting is correct; the is not needed. From the [documentation](https://github.com/apple/swift/blob/9af806e8fd93df3499b1811deae7729176879cb0/stdlib/public/Darwin/Foundation/NSObject.swift#L253):

> when the returned NSKeyValueObservation is deinited or invalidated, it will stop observing

And `drawStatusObservation` (just like any other stored instance property) will be set to `nil` when the instance is deinited